### PR TITLE
 Added functionality to share image directly.

### PR DESCRIPTION
**Pull request by vishwesh3**
#1144 

Implemented for both Android and iOS :)

### CODE REVIEW

**Comment by borisyankov** [comment]
Check the tests.

**Comment by vishwesh3** [comment]
Updated 👍

File: src/api/downloadFile.android.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import type { Auth } from '../types';
 import { getAuthHeader } from '../utils/url';
 import userAgent from '../utils/userAgent';
 
-export default (url: string, auth: Auth) =>
+export default (url: string, auth: Auth, onSuccess?: () => void) =>
```
Comment (borisyankov) [inline src/api/downloadFile.android.js]: Make it an `async` function. Do not use callbacks.

File: src/api/downloadFile.android.js [hunk 2]
```diff
@@ -15,8 +15,12 @@ export default (url: string, auth: Auth) =>
...
-    'User-Agent': userAgent,
-    Authorization: getAuthHeader(auth.email, auth.apiKey),
-  });
+  })
+    .fetch('GET', url, {
+      'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8',
+      'User-Agent': userAgent,
+      Authorization: getAuthHeader(auth.email, auth.apiKey),
+    })
+    .then(res => {
```
Comment (borisyankov) [inline src/api/downloadFile.android.js]: Here you would have just awated the 'res' and then would do a `return res.path()`.

File: src/lightbox/LightboxActionSheet.js [hunk 3]
```diff
@@ -43,9 +43,14 @@ const shareLink = ({ url }: ShareLinkType) => {
...
 };
 
+const shareImageDirectly = ({ url, auth }: DownloadImageType) => {
+  shareImage(url, auth);
+};
+
 const actionSheetButtons: ButtonType[] = [
   { title: 'Download file', onPress: downloadImage },
-  { title: 'Share', onPress: shareLink },
+  { title: 'Share Image', onPress: shareImageDirectly },
```
Comment (borisyankov) [inline src/lightbox/LightboxActionSheet.js]: Fix capitalization as 'Share image'

File: src/lightbox/LightboxActionSheet.js [hunk 4]
```diff
@@ -43,9 +43,14 @@ const shareLink = ({ url }: ShareLinkType) => {
...
+  { title: 'Share Image Link', onPress: shareLink },
```
Comment (borisyankov) [inline src/lightbox/LightboxActionSheet.js]: `Share link to image`?

File: src/lightbox/LightboxActionSheet.js (hunk 3 above)
Comment (vishwesh3) [reply src/lightbox/LightboxActionSheet.js]: Done :+1:

**Comment by vishwesh3** [comment]
Updated 👍

**Comment by borisyankov** [comment]
Looks good 👍

**Pull request merged**